docs/montecarlo/sourceintegration.rst

```diff
@@ -0,0 +1,112 @@
+
+The procedure uses a line absorption rate estimator that is collected during the montecarlo simulation:
+
+.. math::
+
+   \dot E_{lu} = \frac{1}{\Delta t V} \left( 1- e^{-\tau_{lu}}\right) \sum \epsilon
+
+where the sum is over all the packages in a given shell that come into resonance with the transition :math:`u \rightarrow l` during the monte carlo run, :math:`\epsilon` is the energy of one such packet, and :math:`\tau_{lu}` the optical depth of the line.
+
+After the final monte carlo step, a level absorption estimator is calculated, given by:
+
+.. math::
+
+   \dot E_u = \sum_{i < u} \dot E_{iu}
+
+that is, by summing all the line absorption estimators below the currently selected level.
+
+The source function for each line can then be derived from the relation
+
+.. math::
+
+   \left( 1- e^{-\tau_{lu}}\right) S_{ul} = \frac{\lambda_{ul} t}{4 \pi} q_{ul} \dot E_u
+
+where :math:`\lambda_{ul}` is the wavelength of each line  :math:`u \rightarrow l`, and :math:`q_{ul}` is the corresponding branching ratio. The attenuating factor is kept on the left hand side because it is the product of the two that will appear in later formulae.
+
+Having thus produced attenuated source functions from our Monte Carlo run, we move on to using these to calculate the emerging intensity and finally the luminosity per wavelength. The final integral is given as
+
+.. math::
+
+   L_\nu  = 8 \pi^2 \int_0^\infty I_\nu (p) p dp
+
+where :math:`p` is the impact parameter of a ray trough the supernova envelope that reaches the distant observer, and :math:`I_\nu (p)` is the intensity along one such ray, given by recursing trough the list of attenuated source functions from the blue to the red and adding up contributions. The relation linking the intensity before the k:th transition :math:`u \rightarrow l` to the intensity after is
+
+.. math::
+
+   I_k^r = I_k^b e^{-\tau_k} + \left( 1- e^{-\tau_k}\right) S_{k}
+
+where the superscripts are crucial, with :math:`r` and :math:`b` referencing the red and blue sides of the k:th transition respectively. To go from the red side of a line to the blue side of the next we can either ignore continuum sources of opacity, in which case
+
+.. math::
+
+   I_{k+1}^b = I_k^r
+
+.. note::
+
+   Currently the code does not perform the steps necessary to include continuum sources of opacity.
+
+or include them, then requiring we perform
+
+.. math::
+
+   I_{k+1}^b = I_k^r + \Delta \tau_k \left[ \frac 1 2(J_k^r + J_{k+1}^b) - I_k^r  \right]
+
+The starting condition for the blue to red side transition is either
+:math:`I_0^r = B_\nu(T)` if the ray intersects the photosphere and :math:`I_0^r = 0` otherwise.
+
+We seek to integrate all emissions at a certain wavelength :math:`\nu` along a
+ray with impact parameter :math:`p`. Because the supernova ejecta is expanding
+homologously, along any ray parallel to the line of sight, the Doppler effect
```

Comment:
the co-moving frame frequency is continuously shifted to longer wavelength due to the relativistic Doppler effect as the packet/photon propagates.